### PR TITLE
feat: allow error handling from legacy client

### LINKED ISSUE #103: allow to pass a function to handle errors from legacy calls
We have still some places where we use the legacy client and there's no way to handle and catch errors from API calls.
We should pass a function to handle errors if needed and throw the error if there's no "handlers".

### PR DESCRIPTION
**Pull request by Galimede**
Fixes #103

### CODE REVIEW

File: esm/legacy-client.browser.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { initLegacyClient } from './api/v2/legacy-client.js';
 import { prefixUrl } from './prefix-url.js';
 import { request } from './request.fetch.js';
 
-export function initCleverClient (config) {
+export function initCleverClient (config, onError) {
```
Comment (florian-sanders-cc) [inline esm/legacy-client.browser.js:8]: **question:** nothing wrong here but could you provide some context about why we may need this now? (an example of what it allows)

The code itself is ok for me, but I'm missing the point.

**Review by florian-sanders-cc — APPROVED** [review]
Gg @Galimede LGTM, I've just added a small question but no request for changes on my side :wink:

File: esm/legacy-client.browser.js (hunk 1 above)
Comment (Galimede) [reply esm/legacy-client.browser.js:8]: It allows to pass a function to handle errors when there's an API call from the legacy client. We need to do it this way as we don't have another way to handle it after the init.

Comment (pdesoyres-cc) [reply esm/legacy-client.browser.js:8]: Maybe, a little bit of JsDoc may help to understand what to pass here: an optional function

**Review by pdesoyres-cc — APPROVED** [review]
LGTM apart a small comment (do whatever you want with it).

File: esm/legacy-client.browser.js [hunk 2]
```diff
@@ -28,7 +28,15 @@ export function initCleverClient (config) {
...
         .then(addOauthHeader(config))
-        .then(request);
+        .then(request)
+        .catch((error) => {
+          if (onError != null) {
+            return onError(error);
+          }
+          else {
+            throw error;
+          }
```
Comment (hsablonniere) [inline esm/legacy-client.browser.js:38]: nitpick: we can drop the else as the if contains a return, as you want :wink:

**Review by hsablonniere — APPROVED** [review]
Thanks @Galimede for this, you rock :+1: 

I added a small nitpick and I agree with @pdesoyres-cc on the JSDoc comment he made. Apart from that, LGTM :rocket:

**Pull request merged**